### PR TITLE
Refactor some clippy lints

### PR DESCRIPTION
**Pull request by tranzystorek-io**

### CODE REVIEW

File: src/type_.rs [hunk 1]
```diff
@@ -1536,7 +1533,7 @@ pub fn register_import(
 
             // Determine local alias of imported module
             let module_name = match &as_name {
-                None => module[module.len() - 1].clone(),
+                None => module.last().unwrap().clone(),
```
Comment (lpil) [inline src/type_.rs:1536]: Let's use `.gleam_expect` here.

Actually it would be good to disallow use of unwrap if possible.

**Review by lpil — COMMENTED** [review]
Nice! Thank you

Can we remove the use of `bool::then`? I think this is unclear to people less familiar with Rust.

File: src/type_.rs (hunk 1 above)
Comment (tranzystorek-io) [reply src/type_.rs]: Done

**Comment by tranzystorek-io** [comment]
Removed `bool::then` as requested ✅

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**